pkg/utils/gwlog/gwlog.go

```diff
@@ -1,14 +1,120 @@
 package gwlog
 
 import (
+	"context"
+	"fmt"
 	"log"
 	"os"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-type Logger = *zap.SugaredLogger
+type TracedLogger struct {
```

Comment:
So I was thinking it might be useful to see the log lines not linked to any traces (so trace_id=nil). We could use this to drive down log lines without the additional tracking. This would also simplify the conditionals below, so we would basically always append trace_id and whatever the value for it, even if it's missing. What do you think?

Alternatively, let's save the value from `GetTrace()` each time so we don't have to calcluate it twice.

Reply:
I think it depends on what logging tooling you are using. I think in our case we can do something like `-trace_id` to grab all log lines without the trace_id value

I can add it if you think it's useful to others

Reply:
I think it's fine, let's just go with saving the value for now.